description/src/transformer.rs

```diff
@@ -70,3 +75,46 @@ where
         Ok(r)
     }
 }
+
+///
+/// ## Background:
+///
+/// There is a problem in generating recursive type descriptions:
+/// Suppose we have the following setup:
+/// ```rust
+/// struct A {
+///     bees: Vec<B>
+/// }
+///
+/// struct B {
+///     id: u8,
+///     others: Vec<B>
```

Comment:
This is not entirely a valid syntax, does the same problem happen for this? 

```rust
struct B {
  id: u8,
  other: Vec< Box<B> >,
}
```

Reply:
@lexnv What do you mean by "This is not entirely a valid syntax"?

Reply:
Rust would reject recursive types because their `size_of` in bytes wouldn't be known at compile time. And the solution would be to box them, since a pointer always has a known size, depending on the architecture.

I meant that the struct should probably have a boxed::Box(B) instead of recursive type B 😄

Reply:
Ahh I see, its my bad. I was reading it in my mind `struct B { other: B }`. Vec<B> is fine indeed